app/services/LocationService.js

```diff
@@ -449,4 +453,70 @@ export default class LocationServices {
       // nav.navigate('LocationTrackingScreen', {});
     });
   }
+
+  static getCrossedPaths() {
+    return new Promise(resolve => {
+      GetStoreData(CROSSED_PATHS).then(dayBin => {
+        dayBin = JSON.parse(dayBin);
+        if (dayBin !== null && dayBin.reduce((a, b) => a + b, 0) > 0) {
+          console.log('Found crossed paths');
+          resolve(true);
+        } else {
+          console.log("Can't find crossed paths");
+          resolve(false);
+        }
+      });
+    });
+  }
+  static getParticpating() {
+    return new Promise(resolve => {
+      GetStoreData(PARTICIPATE, false).then(isParticipating => {
+        resolve(isParticipating);
+      });
+    });
+  }
+
+  static getBackgroundGeoStatus() {
+    return new Promise(resolve => {
+      BackgroundGeolocation.checkStatus(status => {
+        resolve(status);
+      });
+    });
+  }
+
+  static async checkStatus() {
+    const crossedPaths = await this.getCrossedPaths();
+    const particpating = await this.getParticpating();
+
+    if (!particpating) {
+      return {
+        canTrack: false,
+        reason: 'USER_OFF',
+        crossedPaths: crossedPaths,
+      };
+    }
+
+    const status = await this.getBackgroundGeoStatus();
+    if (!status.locationServicesEnabled) {
+      return {
+        canTrack: false,
+        reason: 'LOCATION_OFF',
```

Comment:
make a enum-ish const for these so they can be exported and used elsewhere to prevent magic strings.

```js
const Reason = {
  LOCATION_OFF: 'LOCATION_OFF',
};
```